proto-lens/src/Data/ProtoLens/Encoding.hs

```diff
@@ -53,20 +53,23 @@ parseMessage :: forall msg . Message msg => Parser () -> Parser msg
 parseMessage end = do
     (msg, unsetFields) <- loop def requiredFields
     if Map.null unsetFields
-        then return $ reverseRepeatedFields fields msg
+        then return $ over unknownFields reverse
```

Comment:
why reversed?

Reply:
nevermind. it's immediately afterwards :)

Reply:
This line causes the list to be accumulated in reversed order:
https://github.com/google/proto-lens/pull/129/files/137e2aab090f96cd7d254531d41cd6537b28ca22#diff-f6d9e63733ce67e5fce8bbe122336aa5R64

It's similar to what we do for repeated fields:
https://github.com/google/proto-lens/pull/129/files/137e2aab090f96cd7d254531d41cd6537b28ca22#diff-f6d9e63733ce67e5fce8bbe122336aa5R126

A little bit of a hack, but at least isolated to this file.

Originally we used `Seq` for repeated fields, but it seemed to make the API unnecessarily complicated.  (And right now it's tricky to have a different type just for accumulation during decode.)